### ISSUE TITLE
kube-aws update fail (prompt for etcd interface in use)

### ISSUE THREAD

**Issue by stanleychow1110** [kubernetes-incubator/kube-aws#583]
i created to build K8S cluster with "kube-aws.sh", they can be created and running properly. At first, i updated cluster.yaml to change worker/controller instance type, minsize of worker/controller, all works, but after some days, doing the same change, it will prompt for etcd interface in use, command use and Cloudformation logs as below 

Command use 
    kube-aws render stack
    kube-aws validate --s3-uri s3://xxx
    kube-aws update --s3-uri s3://xxx

CloudFormation logs:
16:22:14 UTC+0800	UPDATE_ROLLBACK_IN_PROGRESS	AWS::CloudFormation::Stack	omo-dev-kubecluster-Controlplane-RFYB0EJVN8CG	The following resource(s) failed to update: [InstanceEtcd1, InstanceEtcd0, InstanceEtcd2].
16:22:12 UTC+0800	UPDATE_FAILED	AWS::EC2::Instance	InstanceEtcd0	Interface: [eni-8220d5f9] in use.
Physical ID:i-02644b74507b4530c
16:22:12 UTC+0800	UPDATE_COMPLETE	AWS::AutoScaling::LaunchConfiguration	ControllersLC	
16:22:12 UTC+0800	UPDATE_FAILED	AWS::EC2::Instance	InstanceEtcd2	Interface: [eni-b526d3ce] in use.
Physical ID:i-063b8697a22a28ac6
16:22:12 UTC+0800	UPDATE_FAILED	AWS::EC2::Instance	InstanceEtcd1	Interface: [eni-bf4a52f5] in use.
16:22:12 UTC+0800	UPDATE_IN_PROGRESS	AWS::AutoScaling::LaunchConfiguration	ControllersLC	Resource creation Initiated
16:22:11 UTC+0800	UPDATE_IN_PROGRESS	AWS::AutoScaling::LaunchConfiguration	ControllersLC	Requested update requires the creation of a new physical resource; hence creating one.
16:22:11 UTC+0800	UPDATE_IN_PROGRESS	AWS::EC2::Instance	InstanceEtcd2	Requested update requires the creation of a new physical resource; hence creating one.
16:22:11 UTC+0800	UPDATE_IN_PROGRESS	AWS::EC2::Instance	InstanceEtcd0	Requested update requires the creation of a new physical resource; hence creating one.
16:22:10 UTC+0800	UPDATE_IN_PROGRESS	AWS::EC2::Instance	InstanceEtcd1	Requested update requires the creation of a new physical resource; hence creating one.

Thanks for your help.

**Comment by mumoshu** [comment]
@stanleychow1110 Hi, thanks for trying kube-aws!
Probably you've tried to update a cluster created with an older version of kube-aws?
Unfortunately, you need to recreate your cluster to upgdate kube-aws for now.
Please see more context and recommendations about it in #455

**Comment by stanleychow1110** [comment]
Hi, Mumoshu, 

Actually, i used the same version v0.9.4 to build and update the cluster. I found that when the cluster first created, i can manage to update it, but after some days, got the error. I will try the latest final version v0.9.5 to see if problem still exist, thanks.

**Comment by mumoshu** [comment]
@stanleychow1110 Thanks for your cooperation 👍

**Comment by ankon** [comment]
For the future record: I just ran into this and investigated a bit what is actually happening here.

The key is "after some days" really: the Etcd EC2 instances refer to the CoreOS AMI to use, and that one is based on the selected channel in `cluster.yaml`. Eventually it changes, and then CloudFormation (rightfully) wants to replace the Etcd EC2 instances to apply the change in the AMI. This fails, because CloudFormation isn't able to re-use the ENIs, which are still attached to the old Etcd instances.

I found this by trying to create a CloudFormation changeset from the stack that `kube-aws update` produced, and reviewing the planned changes.
